### DISCUSSION TITLE
issue with defining function with "conditional"

### DISCUSSION THREAD

**Discussion by qlyan913** [firedrakeproject/firedrake#3737] [Firedrake support]
I am trying to solving an elliptic PDE on a square with length L in 2D. The coefficient function $A(x)$ is defined as 

$A(x)=\frac{1}{1+\sum_{1\leq i,j\leq L-1} g(x-x_{i,j})}$

where $g(x)$ is a function with compact support in $B_s(0)$ with $s$ the radius of a ball and $x_{i,j}$ are some random points inside square. 

In the code, the function A is defined as 
```
x , y= SpatialCoordinate(mesh)
f_sum=0.0
for i in range(0,nn):
   for j in range(0,nn):
      x_center=[i+1+dn1[i*nn+j],j+1+dn2[i*nn+j]]
      f_sum=f_sum + conditional(((x-x_center[0])**2+(y-x_center[1])**2)**0.5>s,0,20*(1-((x-x_center[0])**2+(y-x_center[1])**2)/pow(s,2))**3*(3*((x-x_center[0])**2+(y-x_center[1])**2)+1))
aexpr=1./(1+f_sum)
A = assemble(interpolate(aexpr, FunctionSpace(mesh, 'CG', 5)))
```
When L =10, the coefficient looks like the following:
<img src="https://github.com/user-attachments/assets/c915f811-fe02-414d-aa9e-eca90e787e0c" width="300" height="300">

However, the code gives the following error warning if I choose L=100: 
> Traceback (most recent call last):
> File "/home/yan00082/firedrake/src/ufl/ufl/core/ufl_type.py", line 206, in _inherited_ufl_shape
    return self.ufl_operands[inherit_shape_from_operand].ufl_shape
  File "/home/yan00082/firedrake/src/ufl/ufl/core/ufl_type.py", line 206, in _inherited_ufl_shape
    return self.ufl_operands[inherit_shape_from_operand].ufl_shape
  File "/home/yan00082/firedrake/src/ufl/ufl/core/ufl_type.py", line 206, in _inherited_ufl_shape
    return self.ufl_operands[inherit_shape_from_operand].ufl_shape
  [Previous line repeated 997 more times]
RecursionError: maximum recursion depth exceeded

Any helps are appreciated.

**Comment by colinjcotter** [comment]
I’m guessing that the UFL just got too complicated. A better way to do this would be to interpolate one term at a time to another field and add this on to your A field using +=

**Reply by qlyan913** [reply]
Thanks for the suggestion. When I define interpolate one term at a time to another field and then add it onto A field. It will gives error message: 'Sum' object has no attribute 'ufl_element'. It seems that interpolating A field again could fix it.  But overall it took a bit more time to get the coefficient than the original way when L=10.

**Reply by colinjcotter** [reply]
That doesn’t sound like you are doing what I suggested.

You are adding using UFL, but I am talking about adding the basis coefficients using the += method of Function.
